docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,257 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream methods for downloading files, it is time to introduce you to additional *upstream* methods. Upstream methods refers to methods that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to handle files and version control a repository with Git-like commands.
+
+## `HfApi`
+
+The `HfApi` class is a high level class that wraps around HTTP requests. There are many useful tasks you can accomplish with the `HfApi` class including: 
+
+- List and filter models.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models, and filter them according to a specific language or framework. This can be especially useful for library and organization owners who want to view all the models they own. Use the `list_models` function with the `filter` parameter to search for a model of interest:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List all models.
+>>> api.list_models()
+
+# List only text classification models.
+>>> api.list_models(filter="text-classification")
```

Comment:
Would it be helpful to mention that these filters are also visible on the website (e.g. with a screenshot) so help guide the reader on what types of filters are possible?

<img width="374" alt="Screen Shot 2021-09-23 at 22 38 02" src="https://user-images.githubusercontent.com/26859204/134580477-da70b994-02a6-4a96-a311-463a31794092.png">